src/gen.c

```diff
@@ -0,0 +1,88 @@
+/**
```

Comment:
Oh, a C program !
I completely forgot that this language existed long time ago... :-)

Reply:
yeah hopefully we never have to touch it again, I included it to "show my work"

Reply:
Actually a great idea to commit the tools ! :-)